### ISSUE TITLE
Stencils overwriting each other from different collections

### ISSUE THREAD

**Issue by ivolff** [defold/defold#9123]
Feels there is an issues with stencils overwriting each other from different collections, so I created a test project to demo this.

### Expected behaviour

![image](https://github.com/defold/defold/assets/46387375/bdf54cf9-6171-4b6e-8dfe-ceb97c745df5)

### Actual behaviour

![image](https://github.com/defold/defold/assets/46387375/ed119b0c-59bc-4476-9a32-71b450c368ef)

### Steps to reproduce
I have two gui-s.
In every gui is a stencil-node, and a "stick" node.

![image](https://github.com/defold/defold/assets/46387375/d4e9af0a-8ec1-4813-abe3-f495b4a04f31)
![image](https://github.com/defold/defold/assets/46387375/f3944146-2ee0-455c-9d37-e474b4ad0521)

Every gui is placed in own collection, and downloaded dynamicly with proxy 
![image](https://github.com/defold/defold/assets/46387375/51544bdf-c491-4687-8b79-81fd3d18d3a8)

### Can just build my test project:
https://github.com/ivolff/defold_stencil_example

<hr/>

<table>
<tr><td>Build time</td><td>2024-05-28T14:20:53.141474</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>7247f038fba6f40cda23c5ea2e38e9fde69a2ea6</td></tr>
<tr><td>Defold engine sha</td><td>fd1ad4c17bfdcd890ea7176f2672c35102384419</td></tr>
<tr><td>Defold version</td><td>1.8.1</td></tr>
<tr><td>GPU</td><td>NVIDIA GeForce RTX 3070 Laptop GPU/PCIe/SSE2</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 NVIDIA 551.61</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 11</td></tr>
<tr><td>OS version</td><td>10.0</td></tr>
<table>

**Comment by AGulev** [comment]
Workaround:

set different order for gui nodes, in the example it will be:
```
 local order = 1
function init(self)
	print("gui created")
	msg.post(".","acquire_input_focus")
	gui.set_render_order(order)
	order = order + 1
end
```

**Comment by AGulev** [comment]
For now, this is handled using `render_order`.  
I tried implementing a separate sorting mechanism per world, but it would remove the ability to sort manually.  
To place something in the background, the developer would have to unload and reload the collection, since the collection loading order would affect the sorting — which feels wrong.

Here’s an example of such an implementation: https://github.com/defold/defold/compare/dev...issue-9123

Since we want to keep the system flexible, this approach is a no-go.  
I would suggest using a Lua module that maintains a list of your GUI components along with their sorting logic, allowing you to manage it manually. Something like:

in your `*.gui`
```lua
local guis = require("modules.guis")

function init()
gui.set_render_order(guis.ORDER[guis.POPUP_INFO])
end
```

in your `guis.gui`:
```lua
local M = {}

M.POPUP_INFO = "POPUP_INFO"
M.POPUP_WIN = "POPUP_WIN"
M.GUI_MAIN = "GUI_MAIN"
...
---- order

M.ORDER = {}
M.ORDER[M.GUI_MAIN] = 0
M.ORDER[M.POPUP_WIN] = 4
M.ORDER[M.POPUP_INFO] = 5
...
return M
```

**Comment by AGulev** [comment]
@britzl I think we don't wanna add one more sorting function and make it more complicated, the issue solvable by `gui.set_render_order()`
Please, take a look, what do you think?

**Comment by britzl** [comment]
> [@britzl](https://github.com/britzl) I think we don't wanna add one more sorting function and make it more complicated, the issue solvable by `gui.set_render_order()` Please, take a look, what do you think?

Makes sense! We should add some documentation to mention this in https://defold.com/manuals/gui-clipping/

**Comment by AGulev** [comment]
I'm not even sure how to add it into this documentation since it's not directly related to stencil, more like unpredicted behavior  of any unsorted GUIs (even without stencil the beh. Will be unpredicted until sorting will be clearly set)

**Comment by AGulev** [comment]
I'm closing this issue since render order should be controlled by `gui.set_render_order()`
Let me know if you have more questions related to it

**Comment by AGulev** [comment]
thinking about it more, I can change priority of the render_order and world order, to keep things sortable by code.
I'll try to prepare a PR
